### ISSUE TITLE
S3242 should not suggest change resulting into S4017

### ISSUE THREAD

**Issue by lg2de** [SonarSource/sonar-dotnet#5443]
### Description

S3242 (Method parameters should be declared with base types) should not suggest changes resulting into S4017 (Method signatures should not contain nested generic types).

### Repro steps

```csharp
private void Handler(IReadOnlyDictionary<string, object> data)
{
    var x = data.ToDictionary(x => x.Key, x => x.Value);
}
```

On the property `data` the message "S3242 Consider using more general type 'System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>' instead of 'System.Collections.Generic.IReadOnlyDictionary<string, object>'." is raised.

Applying the suggested change...
```csharp
private void Handler(IEnumerable<KeyValuePair<string, object>> data)
{
    var x = data.ToDictionary(x => x.Key, x => x.Value);
}
```
... the message "S4017 Refactor this method to remove the nested type argument." is raised.

### Expected behavior

No suggestion.

### Actual behavior

S3242 or S4017 is raised. One of these I need to suppress.

### Known workarounds

Suppression

### Related information

* C#/VB.NET Plugins version: 8.36
* Visual Studio version: 2022
* MSBuild / dotnet version: net472
* SonarScanner for .NET version (if used): N/A
* Operating System: Windows 10
